src/main/kotlin/org/rust/ide/template/postfix/PrintlnPostfixTemplate.kt

```diff
@@ -0,0 +1,140 @@
+        private val macroStart = "$macroName!("
+
+        fun createMacro(expressionText: String, addTrailingSemicolon: Boolean = true): RsExpr {
+            val macroExpression = if (fmt == Fmt.None) {
+                psiFactory.createExpression("$macroStart$expressionText)")
+            } else {
+                psiFactory.createExpression("$macroStart\"${fmt.text}\", $expressionText)")
+            }
+            if (addTrailingSemicolon) {
+                macroExpression.add(psiFactory.createSemicolon())
+            }
+            return macroExpression
+        }
+
+        fun replaceWithMacro(expression: PsiElement, addTrailingComma: Boolean = false) {
+            val newElement = expression.replace(createMacro(expression.text, expression.parent is RsBlock))
+            if (addTrailingComma) {
+                newElement.add(psiFactory.createComma())
+            }
+            editor.caretModel.moveToOffset(newElement.endOffset)
+        }
+    }
+
+    override fun expandForChooseExpression(expression: PsiElement, editor: Editor) {
+        if (expression !is RsExpr) return
+
+        val psiFactory = RsPsiFactory(expression.project)
+        val macroCreator = MacroCreator(editor, psiFactory, macroName, Fmt.fromExpr(expression))
+
+        val parent = expression.parent
+        when (parent) {
+            is RsLetDecl -> {
+                val expressionText = parent.pat?.text ?: return
+                addMacroNextToElement(macroCreator.createMacro(expressionText), parent, psiFactory, editor)
+                if (parent.semicolon == null) {
+                    parent.add(psiFactory.createSemicolon())
+                }
+            }
+            is RsMatchArm -> {
+                val matchBody = parent.ancestorStrict<RsMatchBody>() ?: return
+                if (matchBody.containsUnitArm) {
+                    macroCreator.replaceWithMacro(expression, parent.comma == null)
+                } else {
+                    surroundWithBlockExpression(macroCreator.createMacro(expression.text), expression, psiFactory, editor)
+                }
+            }
+            is RsBinaryExpr -> addMacroNextToElement(macroCreator.createMacro(expression.text), parent, psiFactory, editor)
+            else -> macroCreator.replaceWithMacro(expression)
+        }
+    }
+
+    private fun addMacroNextToElement(macro: RsExpr, element: PsiElement, psiFactory: RsPsiFactory, editor: Editor) {
+        val childToContainer = element.closestContainerAncestor ?: return
+        if (childToContainer.second is RsMatchArm) {
+            surroundWithBlockExpression(macro, childToContainer.first, psiFactory, editor)
+        } else {
+            val newElement = childToContainer.second.addAfter(macro, childToContainer.first.getNextNonCommentSibling()?.prevSibling)
+            editor.caretModel.moveToOffset(newElement.endOffset)
+            newElement.add(psiFactory.createNewline())
```

Comment:
```
newElement.add(psiFactory.createNewline())
newElement.ancestorStrict<RsBlock>()?.let {
    CodeStyleManager.getInstance(newElement.project).reformat(it)
}
```
Seems like it solves your problem with `let b = a - 42 * 3.println`. If no, just leave it the way it is, and I'll try to fix this issue later.

Reply:
Great, works like a charm.
Had to use `trimIndent` in a few `let` tests because of this.